lib/azure/azure_interface.rb

```diff
@@ -0,0 +1,76 @@
+#
+# Author::
+# Copyright:: Copyright (c) 2016 Opscode, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'azure/custom_errors'
+
+class Azure
+  class AzureInterface

```

Comment:
Yeah, if there are methods that we're not going to use, its ok to make them "not implemented." Mostly it's just a consistency thing so we have the flexibility to easily integrate with knife-cloud.